doc: update cc_ssh clarify host and auth keys

powersj opened the pull request:
LP: #1827021

nilsmeyer commented on cloudinit/config/cc_ssh.py:
```diff
@@ -9,11 +9,13 @@
-This module handles most configuration for ssh and ssh keys. Many images have
-default ssh keys, which can be removed using ``ssh_deletekeys``. Since removing
-default keys is usually the desired behavior this option is enabled by default.
+This module handles most configuration for ssh and both host and authroized ssh
```
typo: authorized

powersj replied:
fixed, thanks for the review

OddBloke commented on cloudinit/config/cc_ssh.py:
```diff
@@ -9,11 +9,13 @@
-default ssh keys, which can be removed using ``ssh_deletekeys``. Since removing
-default keys is usually the desired behavior this option is enabled by default.
+This module handles most configuration for ssh and both host and authorized ssh
+keys. Many images have default ssh keys, which can be removed using
```
```suggestion
keys. Many images have default host ssh keys, which can be removed using
```

Looking at it, perhaps "Many images" should be the start of a new paragraph, as it doesn't apply to authorized keys at all.

OddBloke commented on cloudinit/config/cc_ssh.py:
```diff
@@ -9,11 +9,13 @@
-default keys is usually the desired behavior this option is enabled by default.
+This module handles most configuration for ssh and both host and authorized ssh
+keys. Many images have default ssh keys, which can be removed using
+``ssh_deletekeys``. This prevents re-use of a private key from an image on
```
```suggestion
``ssh_deletekeys``. This prevents re-use of a private host key from an image on
```

OddBloke commented on cloudinit/config/cc_ssh.py:
```diff
@@ -9,11 +9,13 @@
+This module handles most configuration for ssh and both host and authorized ssh
+keys. Many images have default ssh keys, which can be removed using
+``ssh_deletekeys``. This prevents re-use of a private key from an image on
+multiple machines. Since removing default keys is usually the desired behavior
```
```suggestion
multiple machines. Since removing default host keys is usually the desired behavior
```

powersj commented:
Ok I've updated the repo based on @OddBloke's comments and added definitions for both authorized and host keys. Honestly adding those definitions helped me too. Most of the diff is due to changes in line length.

OddBloke commented on cloudinit/config/cc_ssh.py:
```diff
@@ -9,19 +9,26 @@
+enabled by default.
+
+Authorized keys are a list of public SSH key pairs that are allowed to
+connect to a system.  Keys can be added using the ``ssh_keys`` configuration
```
`ssh_keys` is for host keys, `ssh_authorized_keys` is for authorized keys.

powersj replied:
Doh. well.... maybe I didn't clear it up for myself as much as I thought I had.

powersj commented:
I gave the two sections headers and moved the authorized section up as I felt it would be more relevant to users looking for docs.

OddBloke commented on cloudinit/config/cc_ssh.py:
```diff
@@ -9,43 +9,26 @@
-
-.. note::
-    to specify multiline private keys, use yaml multiline syntax
+Authorized keys are a list of public SSH key pairs that are allowed to
```
```suggestion
Authorized keys are a list of public SSH keys that are allowed to
```

A key pair is the public key and the private key.

OddBloke commented on cloudinit/config/cc_ssh.py:
```diff
@@ -9,43 +9,26 @@
-.. note::
-    to specify multiline private keys, use yaml multiline syntax
+Authorized keys are a list of public SSH key pairs that are allowed to
+connect to a system. Authorized keys for the default user/first user defined
```
I think "first user defined in ``users``" is a touch long for a /. Perhaps a parenthetical:

```
for the default user (or the first user defined in ``users``) can be specified
```

OddBloke replied:
This expanded sentence will make it a little easier for people to key in on this being the thing they are looking for, I think:

```suggestion
connect to a user account on a system. They are stored in `.ssh/authorized_keys` in that account's home directory. Authorized keys for the default user/first user defined
```

powersj replied:
Done - I do think we are now in the realm of re-writting this module's docs :)

OddBloke commented on cloudinit/config/cc_ssh.py:
```diff
@@ -9,43 +9,26 @@
-
-Supported key types for the ``ssh_keys`` and the ``ssh_genkeytypes`` config
-flags are:
+Importing ssh public keys for the default user (defined in ``users``)) is
```
I think this paragraph you've inherited from the existing docs is just plain wrong. It's the only mention of `allow_publish_ssh_keys` in the entire project.

powersj replied:
Agreed, I think I would normally say we could follow up, but I've removed it for now.

powersj commented:
Ready for another review
